src/azstoragetorch/_client.py

```diff
@@ -396,7 +398,7 @@ def _download_from_unknown_blob_size(
             length = self._PARTITIONED_DOWNLOAD_THRESHOLD
         return self._download_with_retries(offset, length)
 
-    def _download_with_retries(self, pos: int, length: int) -> bytes:
+    def _download_with_retries(self, pos: int, length: int) -> bytes:   # type: ignore[return]
```

Comment:
It would be interesting to see if we could remove this type ignore. Mainly if we had a bug where we actually were not returning `bytes` we would not catch that bug through type checking. Instead maybe we should consider raising an exception at the very end to say retries are exhausted (we would need to figure out the correct exception type we want)? I'm also open to other ideas but it seems like mypy is just not figuring out that the end of the method will never be reached.